### PR TITLE
Add a framebuffer clear for w_terrain and w_overmap erasures, fixes most of the screen buffer issues with mixed fonts. Fix some overmap refresh issues as well.

### LINKED ISSUE #16934: SDL interface ghosting when playing with ASCII
The refreshing of the screen bugs still continues on latest SDL (4949). Even after the commits #16416 #16211 #16162.
The issue have been pointed out in #15881 #15983 #16033 #16416 . Many instances of the ghosting have been fixed, but some still persist, as I'll try to show with screenshots. It happens when you are playing with ASCII, apparently while using different fonts for the maps. My fonts.json:

```
    "fontblending" : false,
    "fontwidth" : 8,
    "fontheight" : 16,
    "fontsize" : 16,
    "typeface" : "Terminus",
    "map_fontwidth" : 14,
    "map_fontheight" : 14,
    "map_fontsize" : 12,
    "map_typeface" : "Topaz-8",
    "overmap_fontwidth" : 14,
    "overmap_fontheight" : 14,
    "overmap_fontsize" : 12,
    "overmap_typeface" : "Topaz-8"
```

The first screen shows the ghosting after a save. I moved a couple tiles to make it more evident.
new cdda bug 01

This one is after a prompt. Don't exactly remember which one.
new cdda bug 02

The same as before, after one turn of rain.
new cdda bug 03

One more
new cdda bug 04

And to finish, one with all the tiles updated by walking around, except the player one. This one refreshes only if you display a different screen like '?' that forces the refresh. I'm the black space between the bluebell and the bush.
new cdda bug 05

### PR DESCRIPTION
**Pull request by DanmakuDan**
After fixing a bunch of individual refresh issues in a pending branch, the screen disappearing on vehicle menu exit after a repair operation showed that `werase(w_terrain)` was found to be missing a matching framebuffer clear. Adding the framebuffer clear there fixed the refresh issues, so this PR was simplified. It also fixes the refresh for the uimenu that had to be removed in a previous PR.

Should fix #16934.

### CODE REVIEW

**Comment by DanmakuDan** [comment]
Pyredrid mentioned that the overmap was having similar issues, those are corrected here as well.

**Comment by DLaboratory** [comment]
Tested on my computer. Worked well. Refresh issues are all gone.

**Comment by illi-kun** [comment]
Could you give any example of glitches which are fixed by this PR? I've test sleep menu and it still leaves some blank unrefreshed part of terrain. To reproduce, call sleep menu, press ESC to cancel, see this:

ui-refresh

File: src/sdltiles.cpp [hunk 1]
```diff
@@ -221,8 +221,8 @@ static bool fontblending = false;
 // Used only while fontlist.txt is created.
 static std::set<std::string> *bitmap_fonts;
 
-static std::vector<curseline> oversized_framebuffer;
-static std::vector<curseline> terminal_framebuffer;
+std::vector<curseline> oversized_framebuffer;
```
Comment (kevingranade) [inline src/sdltiles.cpp]: These shouldn't be externally visible.

Comment (DanmakuDan) [inline src/sdltiles.cpp]: Should the buffers have a method that clears a specific one?

**Comment by DanmakuDan** [comment]
Mostly it includes:

- zombie-related interruptions (x spotted, stop activity)
- redraw after a uimenu object is used
- closing vehicle menu after repairs/long term activity

**Comment by kevingranade** [comment]
Agreed with @illi-kun, I don't know what draw artifacts this claims to fix, so I can't test it.

Also please add an enum that enumerates the framefuffers, and a static function that calls invalidate_framebuffer with the right framebugger to sdltiles instead of making the framebuffers and invalidate_framebuffer globally visible.

**Comment by DanmakuDan** [comment]
Is it OK to include `game.h` in `cursesport.cpp`? The farthest I can go with invalidating would be at `werase(w_terrain)`, where the window contents get cleared, but the game window is not explicitly known to `cursesport.cpp`. The fact that the framebuffer object exists for bridging nonstandard font sizes makes it hard to confine the behavior to `sdltiles.cpp` only.

**Comment by DanmakuDan** [comment]
Refer to #17076 for the improved version of this PR.

**Pull request closed without merging**